### PR TITLE
[glsl-in] Allow code after terminator statements

### PR DESCRIPTION
**Pull request by JCapucho**

### CODE REVIEW

File: src/front/glsl/builtins.rs [hunk 1]
```diff
@@ -1418,6 +1418,7 @@ impl MacroCall {
         body: &mut Block,
         args: &mut [Handle<Expression>],
         meta: SourceMetadata,
+        finished: bool,
```
Comment (kvark) [inline src/front/glsl/builtins.rs:1421]: can we add it to `Context` instead of having a separate argument?

File: block_ctx [hunk 2]
```diff
@@ -0,0 +1,883 @@
+BlockContext {
```
Comment (kvark) [inline block_ctx:1]: is this file included accidentally?

File: dis [hunk 3]
```diff
@@ -0,0 +1,47 @@
+; SPIR-V
```
Comment (kvark) [inline dis:1]: same here

File: test-block-ctx/block_ctx.Fragment-main.txt [hunk 4]
```diff
@@ -0,0 +1,113 @@
+BlockContext {
```
Comment (kvark) [inline test-block-ctx/block_ctx.Fragment-main.txt:1]: also a leftover?

**Review by kvark — COMMENTED** [review]
I think we can make it even better. Keeping the flag around in the Context still requires checking for it on each statement. What if we instead just cut away all of the statements added after return? I.e.
```rust
fn parse_block() {
...
if stmt == Return {
  return_statements = Some(statements.len());
}
... 
if let Some(count) = return_statements {
  statements.drain(count..); // just flush the rest of statements out
}
}
```

It seems to me that this would require only a single place being responsible for this, instead of spreading the tentacles of this logic through the code

File: block_ctx (hunk 2 above)
Comment (JCapucho) [reply block_ctx:1]: yeah, sorry accidentally added some test files from the spirv parser PR

**Comment by JCapucho** [comment]
The method you propose would incur more allocations, If that's not a problem in your opinion then I'll change it.

**Comment by kvark** [comment]
Not a problem. We are talking about GLSL code that does weird stuff, it can suffer :)

**Comment by JCapucho** [comment]
My poor baby :(

**Comment by kvark** [comment]
Sorry, I didn't mean GLSL in general. I mean specifically GLSL code that does anything after `return`. Your baby is great!

**Comment by JCapucho** [comment]
@kvark implemented your suggestion

**Review by kvark — APPROVED** [review]

**Pull request merged**